src/main/java/org/jenkinsci/test/acceptance/docker/Docker.java

```diff
@@ -158,6 +122,80 @@ public DockerImage build(Class<? extends DockerContainer> fixture) throws IOExce
         }
     }
 
+    //package scope for testing purposes. Ideally we should encapsulate Docker interactions so they can be mocked
+    // and call public method.
+    void copyDockerfileDirectory(Class<? extends DockerContainer> fixture, DockerFixture f, File dir)
+            throws IOException {
+        String dockerfilePath = resolveDockerfileLocation(fixture, f);
+        copyDockerfileDirectoryFromClasspath(fixture, dockerfilePath, dir);
+    }
+
+    private String resolveDockerfileLocation(Class<? extends DockerContainer> fixture, DockerFixture f) {
+        String prefix = null;
+        if(isSpecificDockerfileLocationSet(f)) {
+            prefix = f.dockerfile();
+        } else {
+            prefix = fixture.getName();
+        }
+        return prefix.replace('.', '/').replace('$', '/');
+    }
+
+    private void copyDockerfileDirectoryFromClasspath(Class<? extends DockerContainer> fixture, String dockerfileLocation, File dir) throws IOException {
+        File jar = null;
+        try {
+            jar = Which.jarFile(fixture);
+        } catch (IllegalArgumentException e) {
+            // fall through
+        }
+
+        if (jar!=null && jar.isFile()) {
+            // files are packaged into a jar/war. extract them
+            dockerfileLocation += "/";
+            copyDockerfileDirectoryFromPackaged(jar, dockerfileLocation, dir);
+        } else {
+            // Dockerfile is not packaged into a jar file, so copy locally
+            copyDockerfileDirectoryFromLocal(dockerfileLocation, dir);
+        }
+    }
+
+    private boolean isSpecificDockerfileLocationSet(DockerFixture f) {
+        return f.dockerfile() != null && !"".equals(f.dockerfile().trim());
```

Comment:
Yes I know but I have find in  my career doing such "awesome" things using reflection. But well let's assume that in this case since it is for testing we won't get this things.